overlord/devicestate/devicemgr.go

```diff
@@ -2476,6 +2476,10 @@ func (m *DeviceManager) EnsureRecoveryKeys() (*client.SystemRecoveryKeysResponse
 		// shouldn't happen as the marker file is under ubuntu-data
 		return nil, fmt.Errorf("cannot ensure recovery keys without any ubuntu-data mount points")
 	}
+	authKeyDir := dataMountPoints[0]
+	if !model.Classic() {
+		authKeyDir = filepath.Join(authKeyDir, "system-data")
+	}
```

Comment:
this feels odd given that we already pass the model to the initial helper, it seems the solution is to define a HostWritableForMode that does the full correct thing. I'm ok landing this for expedience if then do the right longer term thing